.circleci/config.yml

```diff
@@ -95,62 +106,94 @@ jobs:
     steps:
       - checkout-and-merge-to-main
       - setup_sbt
-      - restore_sbt_cache
+      - restore_deps_cache
       - run:
           name: Run tests
           command: sbt test
-      - save_sbt_cache
+      - save_deps_cache
 
   integration-tests:
     machine: true
     description: "Integration tests"
     steps:
       - checkout-and-merge-to-main
       - setup_sbt
-      - restore_sbt_cache
+      - restore_deps_cache
       - run:
           name: Run integration tests
           command: docker-compose -f .circleci/google-pubsub-emulator-docker-compose.yml up -d && sbt samples/It/test
-      - save_sbt_cache
+      - save_deps_cache
 
   tck-tests:
     machine: true
     description: "TCK tests"
     steps:
       - checkout-and-merge-to-main
       - setup_sbt
-      - restore_sbt_cache
+      - restore_deps_cache
       - run:
           name: Run TCK tests
           command: sbt tck/Test/run
-      - save_sbt_cache
+      - save_deps_cache
 
   validate-docs:
     machine: true
     description: "Validate docs"
     steps:
       - checkout-and-merge-to-main
       - setup_sbt
-      - restore_sbt_cache
+      - restore_deps_cache
       - run:
           name: Validate docs
           command: make -C docs
-      - save_sbt_cache
+      - save_deps_cache
+     
+
+  maven-java-tests:
+    machine: true
+    description: "Maven Java (plugin and archetype tests)"
+    steps:
+      - checkout-and-merge-to-main
+      - restore_deps_cache
+      - set-sdk-version
+      - setup_sbt
+      - run: 
+          name: publish m2
+          command: sbt publishM2
+      - run: 
+          name: Build maven-java
+          command: |
+            cd maven-java
+            echo "Running maven-java with SDK version: '$SDK_VERSION'"
+            mvn versions:set -DnewVersion=$SDK_VERSION 
+            mvn verify install
+      - run: 
+          name: Run integration tests for Value Entity Archetype
+          command: |
+            cd maven-java
+            mvn -B archetype:generate -DgroupId=com.example -DartifactId=counter-value-entity -DarchetypeGroupId=com.lightbend -DarchetypeArtifactId=maven-archetype-akkasls -DarchetypeVersion=$SDK_VERSION
+            cd counter-value-entity
+            mvn -B -Dakkaserverless-sdk.version=$SDK_VERSION compile       
+      - run: 
+          name: Run integration tests for Event Sourced Entity Archetype
+          command: |
+            cd maven-java
+            mvn -B archetype:generate  -DgroupId=com.example -DartifactId=counter-event-sourced-entity -DarchetypeGroupId=com.lightbend -DarchetypeArtifactId=maven-archetype-akkasls-event-sourced-entity -DarchetypeVersion=$SDK_VERSION
+            cd counter-event-sourced-entity
+            mvn -B -Dakkaserverless-sdk.version=$SDK_VERSION compile              
```

Comment:
I'm taking a different approach here than the original build GH Action. The original build was deploying the proxy, the user service and then hitting it with curl. 

I confess, that the main reason that I'm not reproducing it here is because it's a little more complicated to do it in CircleCI. I could have tried harder, but at this stage I'm taking a shortcut. I prefer to move faster with the merge of the projects. 

Also, I believe we will be better served if we use the integration test testkit with a fully implement application and this is something I would like to add once we move the codegen under the SDK project.